site/src/_pages/spec/1_block-types.mdx

```diff
@@ -153,6 +154,27 @@ Note that this is about translating between different JSON Schemas, and is not t
 
 </Frame>
 
+#### JSON Schema extensions
+
+As well as `entityTypeId`, entity schemas SHOULD specify the following field at the root of the schema:
+
+- `labelProperty`: a single `string`, which MUST be present as a key on the schema's `properties`, i.e. it must name a property available on the entity. This will be used by blocks as a display name for the entity where appropriate.
+
+```json
+{
+  ...
+  "properties": {
+    "name": {
+      "type": "string"
+    }
+  },
+  "labelProperty": "name"
+  ...
+}
+```
+
+We will publish a JSON schema vocabulary which allows for validating these custom fields.
```

Comment:
I think we would need a meta-schema to make programatic validation possible, so may be worth mentioning:
```suggestion
We will publish a JSON schema vocabulary and meta-schema which allows for validating these custom fields.
```